include/extractor/guidance/turn_lane_handler.hpp

```diff
@@ -33,27 +33,75 @@ namespace lanes
 {
 class TurnLaneHandler
 {
+    typedef enum TurnLaneScenario {
+        SIMPLE,             // a straightforward assignment
+        PARTITION_LOCAL,    // an assignment that requires partitioning, using local turns
+        SIMPLE_PREVIOUS,    // an assignemtnn using the turns specified at the previous road (e.g.
+                            // traffic light, lanes not drawn up to the intersection)
+        PARTITION_PREVIOUS, // a set of lanes on a turn with a traffic island. The lanes for the
+                            // turn end at the previous turn (parts of it remain valid without being
+                            // shown again)
+        SLIPROAD, // Sliproads are simple assignments that, for better visual representation should
+                  // include turns from other roads in their listings
+        MERGE,    // Merging Lanes
+        NONE,     // not a turn lane scenario at all
+        INVALID,  // some error might have occurred
+        UNKNOWN,  // UNKNOWN describes all cases that we are currently not able to handle
+        NUM_SCENARIOS
+    } TurnLaneScenario;
+
+    const constexpr static char *scenario_names[TurnLaneScenario::NUM_SCENARIOS] = {
+        "Simple",
+        "Partition Local",
+        "Simple Previous",
+        "Partition Previous",
+        "Sliproad",
+        "Merge",
+        "None",
+        "Invalid",
+        "Unknown"};
+
   public:
     typedef std::vector<TurnLaneData> LaneDataVector;
 
     TurnLaneHandler(const util::NodeBasedDynamicGraph &node_based_graph,
-                    const std::vector<std::uint32_t> &turn_lane_offsets,
-                    const std::vector<TurnLaneType::Mask> &turn_lane_masks,
-                    const TurnAnalysis &turn_analysis);
+                    std::vector<std::uint32_t> &turn_lane_offsets,
+                    std::vector<TurnLaneType::Mask> &turn_lane_masks,
+                    LaneDescriptionMap &lane_description_map,
+                    const std::vector<QueryNode> &node_info_list,
+                    const TurnAnalysis &turn_analysis,
+                    LaneDataIdMap &id_map);
+
+    ~TurnLaneHandler();
 
     OSRM_ATTR_WARN_UNUSED
-    Intersection assignTurnLanes(const NodeID at,
-                                 const EdgeID via_edge,
-                                 Intersection intersection,
-                                 LaneDataIdMap &id_map) const;
+    Intersection assignTurnLanes(const NodeID at, const EdgeID via_edge, Intersection intersection);
 
   private:
+    unsigned *count_handled;
+    unsigned *count_called;
```

Comment:
Why pointers?

The raw pointers here and the `new` / `delete` below are a huge red flag to me.